File: DefiLinkEmulator/DefiArduinoEmulator_WebApp/Models/DefiCOMEmulatorStatus.cs
using System;
using System.Collections.Generic;

namespace DefiArduinoEmulator_WebApp.Models
{
    public class DefiCOMEmulatorStatus
    {
        public Dictionary<string, int> DefiCOMParameter {get; set;} = new Dictionary<string, int>();

        public string COMPortName {get; set;}

        public bool IsRunning {get; set;}
    }
}

File: DefiLinkEmulator/DefiArduinoEmulator_WebApp/src/model/DefiCOMEmulatorStatus.ts
export enum DefiParameterCode
{
    Manifold_Absolute_Pressure,
    Engine_Speed,
    Oil_Pressure,
    Fuel_Rail_Pressure,
    Exhaust_Gas_Temperature,
    Oil_Temperature,
    Coolant_Temperature
};

export class DefiCOMEmulatorStatus
{
    public DefiCOMParameter : {[code : string]: number;};
    public COMPortName : string;
    public IsRunning : boolean;
}